WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -637,18 +642,18 @@ extension MediaCoordinator {
     // `Documents` folder.
     // We want to collect more data about that, so we're going to log that info to Sentry,
     // and also delete the `Media` object, since there isn't really a reasonable way to recover from that failure.
-    func addObserverForDeletedFiles(for media: Media) {
-        addObserver({ [weak self] (media, _) in
+    func addObserverForDeletedFiles() {
+        addObserver({ (media, _) in
             guard let mediaError = media.error,
                 media.hasMissingFileError else {
                 return
             }
 
+            self.cancelUploadAndDeleteMedia(media)
```

Comment:
Is there a significance to the change of sequence in here? Originally, this went after the `CrashLogging.logError` call. Just asking in case I missed something.

Reply:
honestly, abundance of caution — when i was banging my head against the wall trying to figure out what was going on here (before figuring out the race condition), i was thinking that maybe calling out to `logError` somehow interrupts the flow of the function _automgaically_, so in desperate attempt i switched them around, but it shouldn't have any impact